edxval/api.py

```diff
@@ -395,6 +395,29 @@ def create_or_update_video_transcript(video_id, language_code, metadata, file_da
     return video_transcript.url()
 
 
+def update_transcript_provider(video_id, language_code, provider):
+    """
+    Update transcript provider for an existing transcript.
+
+    Arguments:
+        video_id: id identifying the video to which the transcript is associated.
+        language_code: language code of a video transcript.
+        provider: transcript provider
+    """
+    video_transcript = VideoTranscript.get_or_none(video_id, language_code)
+
+    if provider and provider not in list(dict(TranscriptProviderType.TRANSCRIPT_MODEL_CHOICES).keys()):
+        raise InvalidTranscriptProvider(f'{provider} transcript provider is not supported')
+
+    if video_transcript:
```

Comment:
While I know that empty/missing provider has been handled [here](https://github.com/openedx/edx-val/pull/562/files#diff-5cb93930cfbc5ddd56013cd73f08792eaee9fde2b7c8fc4a571bc609b12b665bR230). Should we make sure we are not passing an empty provider in this condition in case another function needs to use it in the future? I don't think [this](https://github.com/openedx/edx-val/pull/562/files#diff-5cb93930cfbc5ddd56013cd73f08792eaee9fde2b7c8fc4a571bc609b12b665bR230) will account for empty provider.